pkg/controllers/provisioning/scheduling/nodeset.go

```diff
@@ -59,15 +69,15 @@ func NewNodeSet(ctx context.Context, constraints *v1alpha5.Constraints, client c
 
 func (s *NodeSet) getDaemons(ctx context.Context, constraints *v1alpha5.Constraints) ([]*v1.Pod, error) {
 	daemonSetList := &appsv1.DaemonSetList{}
-	if err := s.client.List(ctx, daemonSetList); err != nil {
+	if err := s.kubeClient.List(ctx, daemonSetList); err != nil {
 		return nil, fmt.Errorf("listing daemonsets, %w", err)
 	}
 	// Include DaemonSets that will schedule on this node
 	var pods []*v1.Pod
 	for _, daemonSet := range daemonSetList.Items {
-		pod := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
-		if err := constraints.ValidatePod(pod); err == nil {
-			pods = append(pods, pod)
+		p := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
+		if err := constraints.ValidatePod(p); err == nil {
```

Comment:
Just noticed that we're validating this here and again on line 46

Reply:
I see this on 46: `constraints: constraints,`, wrong line number?